# Part : 4

#### First taking en tweets from all the tweets collected

In [1]:
import pandas as pd
import json
import re

In [2]:
df1=pd.read_csv('temptweets.csv')
df1

Unnamed: 0                      created_at  \
0               0  Tue Oct 19 11:49:23 +0000 2021   
1               1  Tue Oct 19 11:49:23 +0000 2021   
2               2  Tue Oct 19 11:49:24 +0000 2021   
3               3  Tue Oct 19 11:49:25 +0000 2021   
4               4  Tue Oct 19 11:49:25 +0000 2021   
...           ...                             ...   
20238       20238  Wed Oct 20 11:19:03 +0000 2021   
20239       20239  Wed Oct 20 11:19:05 +0000 2021   
20240       20240  Wed Oct 20 11:19:14 +0000 2021   
20241       20241  Wed Oct 20 11:19:20 +0000 2021   
20242       20242  Wed Oct 20 11:19:24 +0000 2021   

                                                    text  \
0      RT @ceemeagain: เปรียบเทียบตัว AirPods 3 ที่หน...   
1      RT @Mediavenir: 🍎 FLASH - Apple a dévoilé une ...   
2      RT @symonextdoor: Jsp si ils délirent ou si il...   
3      RT @theapplehub: These are the new AirPods ava...   
4      RT @gizmodojapan: 今年はあみとうvsリチャード、じゃなかった。Apple ...   
...                                                  ...   
20238  Charge it wherever you go - Magnetic Wireless ...   
20239  RT @iPhoneMania_jp: M1、M1 Pro、M1 Maxの仕様まとめ〜コア数...   
20240  RT @iPhoneMania_jp: 新型MacBook Pro、14インチモデルのみUS...   
20241  RT @iPhoneMania_jp: 新型MacBook ProはHDMI 2.1ではなく...   
20242  RT @UnicornAPR: Unleashing the fastest noteboo...   

                                                    user lang  \
0      {'id': 4708274054, 'id_str': '4708274054', 'na...   th   
1      {'id': 1216392206753452032, 'id_str': '1216392...   fr   
2      {'id': 1012971181576802304, 'id_str': '1012971...   fr   
3      {'id': 1082043983528906754, 'id_str': '1082043...   en   
4      {'id': 1314456811370500097, 'id_str': '1314456...   ja   
...                                                  ...  ...   
20238  {'id': 1399273436208254977, 'id_str': '1399273...   en   
20239  {'id': 3186507852, 'id_str': '3186507852', 'na...   ja   
20240  {'id': 3186507852, 'id_str': '3186507852', 'na...   ja   
20241  {'id': 3186507852, 'id_str': '3186507852', 'na...   ja   
20242  {'id': 72325162, 'id_str': '72325162', 'name':...   en   

                                 name  friends_count    screen_name  \
0      ' JJORJINN ||| ไฟนอลอีสคัมมิ่ง            926      _jjin0309   
1             kli, CEO of guignols 🇩🇿            581        CalyMui   
2                              Oli 75            991      oli__lxxv   
3                       branden repke            323  repke_branden   
4                                 UQE            281    UQE_Digiana   
...                               ...            ...            ...   
20238                         Mobee.k              2        mobee_k   
20239                             あいみ             47    Aimi_Sphere   
20240                             あいみ             47    Aimi_Sphere   
20241                             あいみ             47    Aimi_Sphere   
20242                 manuj sharan 🇮🇳            208    manujsharan   

                      location  followers_count  favourites_count  \
0      ชิปทุกคู่ที่ใจอยากจะชิป              113              1444   
1                Paris, France              146              3782   
2                          NaN             2827             38882   
3                United States               39             35178   
4                          福岡県              178              8876   
...                        ...              ...               ...   
20238                      NaN               22               211   
20239                        🌏               79                21   
20240                        🌏               79                21   
20241                        🌏               79                21   
20242                   Mumbai              198            106532   

       listed_count  statuses_count  
0                 2          222367  
1                 0            9851  
2                 0           18889  
3                 0            186

### We will drop user column as not required

In [3]:
del df1["Unnamed: 0"]
del df1["user"]

In [4]:
df1

created_at  \
0      Tue Oct 19 11:49:23 +0000 2021   
1      Tue Oct 19 11:49:23 +0000 2021   
2      Tue Oct 19 11:49:24 +0000 2021   
3      Tue Oct 19 11:49:25 +0000 2021   
4      Tue Oct 19 11:49:25 +0000 2021   
...                               ...   
20238  Wed Oct 20 11:19:03 +0000 2021   
20239  Wed Oct 20 11:19:05 +0000 2021   
20240  Wed Oct 20 11:19:14 +0000 2021   
20241  Wed Oct 20 11:19:20 +0000 2021   
20242  Wed Oct 20 11:19:24 +0000 2021   

                                                    text lang  \
0      RT @ceemeagain: เปรียบเทียบตัว AirPods 3 ที่หน...   th   
1      RT @Mediavenir: 🍎 FLASH - Apple a dévoilé une ...   fr   
2      RT @symonextdoor: Jsp si ils délirent ou si il...   fr   
3      RT @theapplehub: These are the new AirPods ava...   en   
4      RT @gizmodojapan: 今年はあみとうvsリチャード、じゃなかった。Apple ...   ja   
...                                                  ...  ...   
20238  Charge it wherever you go - Magnetic Wireless ...   en   
20239  RT @iPhoneMania_jp: M1、M1 Pro、M1 Maxの仕様まとめ〜コア数...   ja   
20240  RT @iPhoneMania_jp: 新型MacBook Pro、14インチモデルのみUS...   ja   
20241  RT @iPhoneMania_jp: 新型MacBook ProはHDMI 2.1ではなく...   ja   
20242  RT @UnicornAPR: Unleashing the fastest noteboo...   en   

                                 name  friends_count    screen_name  \
0      ' JJORJINN ||| ไฟนอลอีสคัมมิ่ง            926      _jjin0309   
1             kli, CEO of guignols 🇩🇿            581        CalyMui   
2                              Oli 75            991      oli__lxxv   
3                       branden repke            323  repke_branden   
4                                 UQE            281    UQE_Digiana   
...                               ...            ...            ...   
20238                         Mobee.k              2        mobee_k   
20239                             あいみ             47    Aimi_Sphere   
20240                             あいみ             47    Aimi_Sphere   
20241                             あいみ             47    Aimi_Sphere   
20242                 manuj sharan 🇮🇳            208    manujsharan   

                      location  followers_count  favourites_count  \
0      ชิปทุกคู่ที่ใจอยากจะชิป              113              1444   
1                Paris, France              146              3782   
2                          NaN             2827             38882   
3                United States               39             35178   
4                          福岡県              178              8876   
...                        ...              ...               ...   
20238                      NaN               22               211   
20239                        🌏               79                21   
20240                        🌏               79                21   
20241                        🌏               79                21   
20242                   Mumbai              198            106532   

       listed_count  statuses_count  
0                 2          222367  
1                 0            9851  
2                 0           18889  
3                 0            1860  
4                 5            6991  
...             ...             ...  
20238             0           11055  
20239             2           55820  
20240             2           55821  
20241             2           55823  
20242            18           74019  

[20243 rows x 11 columns]

### Now Taking tweets which are in English Language

In [5]:
df2=[]
for tweet in range(len(df1)):
    if df1.lang[tweet]=="en":
        df2.append(df1.loc[tweet,])
df2 = pd.DataFrame(df2) 
df2

created_at  \
3      Tue Oct 19 11:49:25 +0000 2021   
8      Tue Oct 19 11:49:27 +0000 2021   
13     Tue Oct 19 11:49:29 +0000 2021   
15     Tue Oct 19 11:49:29 +0000 2021   
18     Tue Oct 19 11:49:32 +0000 2021   
...                               ...   
20223  Wed Oct 20 11:18:19 +0000 2021   
20225  Wed Oct 20 11:18:21 +0000 2021   
20226  Wed Oct 20 11:18:24 +0000 2021   
20238  Wed Oct 20 11:19:03 +0000 2021   
20242  Wed Oct 20 11:19:24 +0000 2021   

                                                    text lang  \
3      RT @theapplehub: These are the new AirPods ava...   en   
8              For all your quality perfumes\nContact us   en   
13     RT @Fragrance__GH: For quality perfumes, buy f...   en   
15     RT @appleinsider: New #MacBookPro models limit...   en   
18               Blah blah no touchscreen not interested   en   
...                                                  ...  ...   
20223  RT @DigiSliceX:  Apple's AR headset productio...   en   
20225  RT @DigiSliceX:  [LEAKED] Alleged first look ...   en   
20226  Apple 1 Year Limited Warranty with 90 Day Tech...   en   
20238  Charge it wherever you go - Magnetic Wireless ...   en   
20242  RT @UnicornAPR: Unleashing the fastest noteboo...   en   

                           name  friends_count      screen_name  \
3                 branden repke            323    repke_branden   
8         BREAST BOARD CHAIRMAN           2261     NUFUO_mu_IGP   
13        BREAST BOARD CHAIRMAN           2261     NUFUO_mu_IGP   
15     TheTransparentPharmacist            165   TransparentRPH   
18                 photographer           3088  chadlawsonphoto   
...                         ...            ...              ...   
20223                ⚔️Kuneho⚔️            272        panaloTOL   
20225                ⚔️Kuneho⚔️            272        panaloTOL   
20226                     Lynda             16        Lyndauusa   
20238                   Mobee.k              2          mobee_k   
20242           manuj sharan 🇮🇳            208      manujsharan   

                        location  followers_count  favourites_count  \
3                  United States               39             35178   
8                            NaN             3782             12524   
13                           NaN             3782             12524   
15     Somewhere in the South...               33                86   
18                   Atlanta, GA             5906              7755   
...                          ...              ...               ...   
20223                      Pinas               20              2110   
20225                      Pinas               20              2110   
20226                        NaN                0                16   
20238                        NaN               22               211   
20242                     Mumbai              198            106532   

       listed_count  statuses_count  
3                 0            1860  
8                 4           15413  
13                4           15414  
15                2            2266  
18               23          290793  
...             ...             ...  
20223             0            2748  
20225             0            2749  
20226             0              17  
20238             0           11055  
20242            18           74019  

[7284 rows x 11 columns]

# Out of 20,000 we got 7284 tweets

### Cleaning of tweets by removing special characters

In [6]:
def clean_text(df, text_field):
    #df[text_field] = df[text_field].str.lower()
    df[text_field] = df[text_field].apply(lambda elem: re.sub("(RT @\w+:)|(RT+)|([^0-9A-Za-z \t])|(\w+:\/\/\S+)", " ", elem)) 
    return df
df2 = clean_text(df2, 'text')



Display of Data

In [7]:
df2.text

3          These are the new AirPods available for  179...
8                 For all your quality perfumes Contact us
13         For quality perfumes  buy from us     Delive...
15          New  MacBookPro models limited to HDMI 2 0    
18                 Blah blah no touchscreen not interested
                               ...                        
20223        Apple s AR headset production pushed back ...
20225         LEAKED  Alleged first look at the  Beats ...
20226    Apple 1 Year Limited Warranty with 90 Day Tech...
20238    Charge it wherever you go   Magnetic Wireless ...
20242      Unleashing the fastest notebook ever  Loaded...
Name: text, Length: 7284, dtype: object

In [8]:
#save to csv and send for analyser..
df2.to_csv("cleantweets.csv")